runtime/src/main/java/org/corfudb/runtime/view/stream/AbstractQueuedStreamView.java

```diff
@@ -297,20 +299,23 @@ public synchronized ILogData previous() {
                 || prevAddress != null && prevAddress <= context.minResolution) {
             context.globalPointer = prevAddress == null ? Address.NEVER_READ :
                                                 prevAddress - 1L;
-            long oldPointer = context.globalPointer;
+
             remainingUpTo(context.minResolution);
             context.minResolution = Address.NON_ADDRESS;
             context.globalPointer = oldPointer;
             prevAddress = context
                     .resolvedQueue.lower(context.globalPointer);
-            log.trace("Previous[}] updated queue {}", this, context.resolvedQueue);
+            log.trace("previous[{}]: updated resolved queue {}", this, context.resolvedQueue);
         }
+
+        // Clear the read queue, it may no longer be valid
```

Comment:
can we have some sort of picture or ascii art for this. I keep forgetting the whole mechanism and it is hard for me to review it after sometime. Would be nice if we can have a picture.